### PR TITLE
refact (pt_expt): provide infrastructure for converting dpmodel classes to PyTorch modules. 

### PR DESCRIPTION
**Pull request by wanghan-iapcm**
consider after the merge of #5194

automatically wrapping dpmodel classes (array_api_compat-based) as PyTorch modules. The key insight is to detect attributes by their **value type** rather than by hard-coded names. 

## Summary by CodeRabbit

* **New Features**
  * Registry-driven conversion for DP objects to PyTorch modules enabling automatic wrapper creation.
  * New PyTorch-friendly descriptor variants with stable forward outputs for se_e2_a and se_r.
  * PyTorch-wrapped exclude-mask utilities and a NetworkCollection of wrapped network types for proper module/state handling.
  * Device-aware tensor conversion and robust handling of numpy buffers and None-valued buffers for reliable serialization/movement.

### CODE REVIEW

File: deepmd/pt_expt/common.py [hunk 1]
```diff
@@ -0,0 +1,278 @@
...
+this registry to find the appropriate converter.
+
+Examples of registered mappings:
+- AtomExcludeMaskDP → lambda v: AtomExcludeMask(v.ntypes, exclude_types=...)
+- NetworkCollectionDP → lambda v: NetworkCollection.deserialize(v.serialize())
+"""
+
+
+def register_dpmodel_mapping(
+    dpmodel_cls: type, converter: Callable[[Any], torch.nn.Module]
```
Comment (njzjz) [inline deepmd/pt_expt/common.py]: Considering that any dpmodel class is a subclass of `NativeOP`, we could replace `Any` with `NativeOP` in this file.
```suggestion
    dpmodel_cls: type[NativeOP], converter: Callable[[NativeOP], torch.nn.Module]
```

File: deepmd/pt_expt/common.py [hunk 2]
```diff
@@ -0,0 +1,278 @@
...
+    # clear an existing buffer to None
+    if value is None and "_buffers" in obj.__dict__ and name in obj._buffers:
+        obj._buffers[name] = None
+        return True, None
+
+    # dpmodel object → pt_expt module
+    if "_modules" in obj.__dict__:
+        converted = try_convert_module(value)
+        if converted is not None:
+            return False, converted
```
Comment (njzjz) [inline deepmd/pt_expt/common.py]: Do we need to have a check here? For example, if an object is a `NativeOP` but not a `torch.nn.Module`, we think it can be converted. But if it is not in the map, we consider that something went wrong.

**Review by njzjz — APPROVED** [review]
